package/MDAnalysis/analysis/base.py

```diff
@@ -407,49 +730,121 @@ def run(self, start=None, stop=None, step=None, frames=None,
             non-default value will raise a :exc:`ValueError`.
 
-            .. versionadded:: 2.2.0
-
+        .. versionadded:: 2.2.0
         verbose : bool, optional
             Turn on verbosity
-        
+
         progressbar_kwargs : dict, optional
-            ProgressBar keywords with custom parameters regarding progress bar position, etc; 
+            ProgressBar keywords with custom parameters regarding progress bar position, etc;
             see :class:`MDAnalysis.lib.log.ProgressBar` for full list.
-
+            Available only for `backend='local'`
+
+        backend : str, optional
+            Enables parallel execution of :meth:`AnalysisBase.run`. To list all available backends for your class,
+            check `self.available_backends` attribute.
+        n_workers : int, optional
+            number of workers to split work across.
+        n_parts : int, optional
+            number of parts to split computations across. Can be more than number of workers,
+            especially if you want to get a progressbar with `dask.distributed` backend.
+        client : dask.distributed.Client, optional
+            if running 'dask.distributed' backend, you must pre-configure a Client object, and then pass it
+            as an argument to execute `run()` method on this Client. Note that in this case, `n_workers`
+            will be ignored -- all workers from Client object will be used.
 
         .. versionchanged:: 2.2.0
             Added ability to analyze arbitrary frames by passing a list of
             frame indices in the `frames` keyword argument.
 
         .. versionchanged:: 2.5.0
-            Add `progressbar_kwargs` parameter, 
+            Add `progressbar_kwargs` parameter,
             allowing to modify description, position etc of tqdm progressbars
         """
-        logger.info("Choosing frames to analyze")
-        # if verbose unchanged, use class default
-        verbose = getattr(self, '_verbose',
-                          False) if verbose is None else verbose
+        # default to local execution
+        backend = 'local' if backend is None and client is None else backend
 
-        self._setup_frames(self._trajectory, start=start, stop=stop,
-                           step=step, frames=frames)
-        logger.info("Starting preparation")
+        if progressbar_kwargs and ((client is not None) or (backend not in (None, "local"))):
+            raise NotImplementedError("Can not display progressbar with non-local backend")
+
+        if n_workers is None:
+            if client:
+                n_workers = sum(client.nthreads().values())
+            else:
+                n_workers = 1
+
+        # validate input parameters
+        n_parts = n_workers if n_parts is None else n_parts
+
+        # do this as early as possible to check client parameters before any computations occur
+        client = self._configure_client(backend=backend, n_workers=n_workers, client=client)
```

Comment:
`client` is both a `ParallelExecutor` and a `dask.distributed.Client` which is confusing.

Reply:
done